src/Microsoft.Data/DataFrameTable.cs

```diff
@@ -147,5 +147,82 @@ public int GetColumnIndex(string columnName)
             return -1;
         }
 
+        public void AppendRow(long index, string[] values)
+        {
+            if (values.Length != ColumnCount)
+            {
+                throw new ArgumentException($"Expected values.Length {values.Length} to be the number of columns in the table {ColumnCount}");
+            }
+            for (int i = 0; i < values.Length; i++)
+            {
+                string value = values[i];
+                BaseColumn column = Column(i);
+                bool parse;
+                switch (column.DataType)
+                {
+                    case Type boolType when boolType == typeof(bool):
+                        parse = bool.TryParse(value, out bool boolAppend);
+                        (column as PrimitiveColumn<bool>).Append(parse ? boolAppend : default);
```

Comment:
It feels wrong to me that the DataFrameTable is responsible for parsing these values. It feels more tied to the CSV reading implementation.